pkg/asset/installconfig/aws/validation.go

```diff
@@ -0,0 +1,153 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/ipnet"
+	"github.com/openshift/installer/pkg/types"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+)
+
+// Validate executes platform-specific validation.
+func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	if config.Platform.AWS != nil {
+		allErrs = append(allErrs, validatePlatform(ctx, meta, field.NewPath("platform", "aws"), config.Platform.AWS, config.Networking, config.Publish)...)
+
+		if config.ControlPlane != nil && config.ControlPlane.Platform.AWS != nil {
+			allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS)...)
+		}
+		if len(config.Compute) > 0 {
+			fldPath := field.NewPath("compute")
+			for idx, compute := range config.Compute {
+				fldPath := fldPath.Index(idx)
+				if compute.Platform.AWS != nil {
+					allErrs = append(allErrs, validateMachinePool(ctx, meta, fldPath.Child("platform", "aws"), config.Platform.AWS, compute.Platform.AWS)...)
+				}
+			}
+		}
+	}
+	return allErrs.ToAggregate()
+}
+
+func validatePlatform(ctx context.Context, meta *Metadata, fldPath *field.Path, platform *awstypes.Platform, networking *types.Networking, publish types.PublishingStrategy) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(platform.Subnets) > 0 {
+		allErrs = append(allErrs, validateSubnets(ctx, meta, fldPath.Child("subnets"), platform.Subnets, networking, publish)...)
+	}
```

Comment:
Also:

```go
if platform.DefaultMachinePlatform != nil {
  allErrs = append(allErrs, validateMachinePool(ctx, meta, fldPath.Child("defaultMachinePlatform"), platform, platform.DefaultMachinePlatform)...)
}
```
or some such.  Are we comfortable erroring out if even if the default platform is completely masked by explicit per-pool platform configs?  I am, but I could see how that could go either way.

Reply:
Yeah, let me add default validation, and we can make smarter validations if we see it affecting users and followup.

Reply:
@wking fixed in https://github.com/openshift/installer/compare/a96182f3f4d7c7b69bd674bd918dd72a532f6ce4..9ce0bb39f5cbc2d581705794c8e16933cae350d4